slsDetectorSoftware/src/slsDetector.cpp

```diff
@@ -2666,6 +2677,143 @@ uint32_t slsDetector::getADCEnableMask() {
     return shm()->adcEnableMask;
 }
 
+void slsDetector::setADCInvert(uint32_t value) {
+    uint32_t arg = value;
+    FILE_LOG(logDEBUG1) << "Setting ADC Invert to 0x" << std::hex << arg << std::dec;
+    if (shm()->onlineFlag == ONLINE_FLAG) {
+        sendToDetector(F_SET_ADC_INVERT, &arg, sizeof(arg), nullptr, 0);
+    }
+}
+
+uint32_t slsDetector::getADCInvert() {
+    uint32_t retval = -1;
+    FILE_LOG(logDEBUG1) << "Getting ADC Invert";
+    if (shm()->onlineFlag == ONLINE_FLAG) {
+        sendToDetector(F_GET_ADC_INVERT, nullptr, 0, &retval, sizeof(retval));
+        FILE_LOG(logDEBUG1) << "ADC Invert: 0x" << std::hex << retval << std::dec;
+    }
+    return retval;
+}
+
+int slsDetector::setExternalSamplingSource(int value) {
+    int arg = value;
+    int retval = -1;
+    FILE_LOG(logDEBUG1) << "Setting External Sampling Source to " << arg;
+    if (shm()->onlineFlag == ONLINE_FLAG) {
+        sendToDetector(F_EXTERNAL_SAMPLING_SOURCE, arg, retval);
+        FILE_LOG(logDEBUG1) << "External Sampling source: " << retval;
+    }
+    return retval;
+}
+
+int slsDetector::getExternalSamplingSource() {
+    return setExternalSamplingSource(-1);
+}
+
+int slsDetector::setExternalSampling(int value) {
+    int arg = value;
+    int retval = -1;
+    FILE_LOG(logDEBUG1) << "Setting External Sampling to " << arg;
+    if (shm()->onlineFlag == ONLINE_FLAG) {
+        sendToDetector(F_EXTERNAL_SAMPLING, arg, retval);
+        FILE_LOG(logDEBUG1) << "External Sampling: " << retval;
+    }
+    return retval;
+}
+
+int slsDetector::getExternalSampling() {
+    return setExternalSampling(-1);
+}
+
+void slsDetector::setReceiverDbitList(std::vector<int> list) {
+    FILE_LOG(logDEBUG1) << "Setting Receiver Dbit List";
+
+    if (list.size() > 64) {
+        throw sls::RuntimeError("Dbit list size cannot be greater than 64\n");
+    }
+    for (auto &it : list) {
+        if (it < 0 || it > 63) {
+            throw sls::RuntimeError("Dbit list value must be between 0 and 63\n");
+        }
+    }
+
+    // copy size and vector to shm
+    shm()->rxDbitListSize = list.size();
+    std::copy(list.begin(), list.end(), shm()->rxDbitList);
+ 
+    int ret = FAIL;
+    if (shm()->rxOnlineFlag == ONLINE_FLAG) {
+        int fnum = F_SET_RECEIVER_DBIT_LIST;
+        int arg = list.size();
+        auto receiver =
+            sls::ClientSocket("Receiver", shm()->rxHostname, shm()->rxTCPPort);
```

Comment:
I think it woudl be better to use the normal interface, meaning sendToReceiver? Is there any technical reason why you don't want to do this?

Reply:
I used it as you suggested. I can't do it for the F_GET_RECEIVER_DBIT_LIST, as I need to get the size first before I get the array.